shared/js/pageStore.js

```diff
@@ -63,8 +71,15 @@ loop.store.PageStore = function(mozL10n) {
 
     /**
      * Handle AddedPage action by updating the store state.
+     *
+     * @note If actionData.url already exists in the store, it will not
+     * be added a second time.
      */
     addedPage(actionData) {
+      if (this._hasPageForUrl(actionData.url)) {
+        return;
+      }
```

Comment:
Because fetching the metadata takes so long, if the link is opened quickly on the guest side, they can race and the link-clicker will see it twice.  I'll add a comment.